kratos/solving_strategies/convergencecriterias/displacement_criteria.h

```diff
@@ -124,93 +114,110 @@ class DisplacementCriteria : public ConvergenceCriteria< TSparseSpace, TDenseSpa
     ///@name Operators
     ///@{
 
-    /*Criterias that need to be called after getting the solution */
+    /**
+     * Compute relative and absolute error.
+     * @param rModelPart Reference to the ModelPart containing the problem.
+     * @param rDofSet Reference to the container of the problem's degrees of freedom (stored by the BuilderAndSolver)
+     * @param A System matrix (unused)
+     * @param Dx Vector of results (variations on nodal variables)
+     * @param b RHS vector (residual + reactions)
+     * @return true if convergence is achieved, false otherwise
+     */
     bool PostCriteria(
         ModelPart& rModelPart,
         DofsArrayType& rDofSet,
         const TSystemMatrixType& A,
         const TSystemVectorType& Dx,
         const TSystemVectorType& b
-    ) override
+        ) override
     {
-        if (SparseSpaceType::Size(Dx) != 0) //if we are solving for something
-        {
-            TDataType mFinalCorrectionNorm = TSparseSpace::TwoNorm(Dx);
+        const TDataType tolerance = std::numeric_limits<TDataType>::epsilon();
+        const SizeType size_Dx = Dx.size();
+        if (size_Dx != 0) { //if we are solving for something
+            SizeType size_solution;
+//             TDataType final_correction_norm = TSparseSpace::TwoNorm(Dx); // NOTE: This doesn't take into account the fix dofs
+            TDataType final_correction_norm = CalculateFinalCorrectionNorm(size_solution, rDofSet, Dx);
 
-            TDataType ratio = 0.00;
+            TDataType ratio = 0.0;
 
             CalculateReferenceNorm(rDofSet);
 
-            if(mFinalCorrectionNorm == 0)
-            {
+            if(final_correction_norm < tolerance) {
                 ratio = 0.0;
+            } else {
+                KRATOS_ERROR_IF(mReferenceDispNorm < tolerance) << "NaN norm is detected" << std::endl;
```

Comment:
i think this does happen. I wouldn't throw an error here...

Reply:
Ok